heimctl/retention/scanner.go

```diff
@@ -0,0 +1,80 @@
+package retention
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"euphoria.io/heim/backend/cluster"
+	"euphoria.io/heim/backend/psql"
+	"euphoria.io/scope"
+)
+
+const (
+	maxErrors   = 3
+	GracePeriod = time.Hour
+)
+
+func scanForExpired(ctx scope.Context, c cluster.Cluster, pb *psql.Backend) error {
+	rows, err := pb.DbMap.Select(
+		psql.Room{},
+		"SELECT name, founded_by, retention_days FROM room WHERE retention_days > 0")
+	if err != nil {
+		return err
+	}
+	for _, row := range rows {
+		room, ok := row.(*psql.Room)
+		if !ok {
+			fmt.Printf("error: expected row of type *psql.Room, got %T\n", row)
+			continue
+		}
+		var oldest time.Time
```

Comment:
Selecting directly into a Time doesn't work for me (maybe something changed in gorp?) To fix, I changed this into:

```
var row struct {
  Oldest time.Time
}
```